modules/bam2fastq/1.1/config/default.yaml

```diff
@@ -17,7 +17,7 @@ lcr-modules:
             bam2fastq: 4
 
         mem_mb:
-            bam2fastq: 5000
+            bam2fastq: 100000
```

Comment:
This alone might be justification for switching to the samtools equivalent of this instead of using Picard. I suspect the limitation with using samtools is that the bam would need to be re-sorted on read name first.

Reply:
I agree. I tested samtools on handful of bams that were giving errors with picard, and samtools worked much faster and did not need that much memory. I did not need to re-sort, however that might not be the case for other samples. As I we are going to use reference fasta file from the header, it might be easier to use samtools instead of picard

Reply:
How can samtools export read 1 and read 2 in order if the bam is not readsorted? I thought Picard did this in memory (hence the huge RAM burden)

Reply:
Yes, samtools need bam files to be sorted. I thought that the bam files we have are already sorted.

Reply:
There are two ways to sort a bam file. Read name or position. All ours are sorted on position. I suspect they would need to be re-sorted on read name

Reply:
It is true, they will need to be resorted by position. I think I will modify the whole bam2fastq_run rule and use samtools instead, with read-sorting them in advance.

Reply:
They would need to be sorted by read name. If you tackle this I suggest you and I chat about how to get samtools to sort efficiently. The default produces a lot of small tempfiles and causes a lot of I/O

Reply:
I am also slightly concerned there's a fundamental reason why samtools wasn't the first choice. Maybe we should ask Bruno on Monday